### ISSUE TITLE
A reason that hinders the AI

### ISSUE THREAD

**Issue by Akatsuki57** [stefanhendriks/Dune-II---The-Maker#497]
![800x600_0000](https://user-images.githubusercontent.com/99535165/156874195-c46286c2-9f37-41f2-8d6c-e967d4cb21bb.png)

I played d2tm a lot and watched the game itself.

Very often I come across cases like the following photo shows.

Buildings block each other. I know carry-alls drop units but Harvesters don't. The AI ​​has built 3 refineries and only 2 are functional.

Can't we improve this?

Along the same lines, it's frustrating not being able to build a building one square away from another.

**Comment by stefanhendriks** [comment]
Yes, this is a known bug. What we can do to prevent this is mark cells around structures which are 'free'. Although this will not solve everything, it will prevent the above situation. It won't solve the problem with like 3 cells in a row (but neither are accessible from outside).

About placing a structure one cell away. That would be an 'optional' thing. You can achieve such a thing btw by placing slabs.

**Comment by Akatsuki57** [comment]
Placing structures on a cell is a gameplay enhancement. I find that it is not "practical" to anticipate the construction of slabs before, especially when you want to place a building quickly.
